### PR TITLE
Add support for system metadata

### LINKED ISSUE #915: system metadata
- add name #914 
 - description
 - add all this to another file like metatadata.json
 - zip all the system files into system name
 - may want to make considerations for scenarios, shared time series files, and other situations.

### PR DESCRIPTION
**Pull request by daniel-thom**
Fixes #915 
Dependent on https://github.com/NREL-SIIP/InfrastructureSystems.jl/pull/302 getting merged and tagged.

### CODE REVIEW

**Comment by daniel-thom** [comment]
Here is an example metadata file that gets produced through `to_json(sys, "sys.json")`:

```
{
    "name": "test_rts",
    "description": "test RTS system",
    "frequency": 60,
    "time_series_resolution_minutes": 60,
    "component_counts": [
        {
            "type": "Arc",
            "count": 109
        },
        {
            "type": "Area",
            "count": 3
        },
        {
            "type": "Bus",
            "count": 73
        },
        {
            "type": "GenericBattery",
            "count": 1
        },
        {
            "type": "HVDCLine",
            "count": 1
        },
        {
            "type": "HydroDispatch",
            "count": 1
        },
        {
            "type": "HydroEnergyReservoir",
            "count": 19
        },
        {
            "type": "Line",
            "count": 105
        },
        {
            "type": "LoadZone",
            "count": 3
        },
        {
            "type": "PowerLoad",
            "count": 51
        },
        {
            "type": "RenewableDispatch",
            "count": 30
        },
        {
            "type": "RenewableFix",
            "count": 31
        },
        {
            "type": "TapTransformer",
            "count": 15
        },
        {
            "type": "ThermalStandard",
            "count": 76
        },
        {
            "type": "VariableReserve",
            "count": 2
        }
    ],
    "time_series_counts": [
        {
            "type": "DeterministicSingleTimeSeries",
            "count": 182
        },
        {
            "type": "SingleTimeSeries",
            "count": 182
        }
    ],
    "user_data": {
        "author": "dan"
    }
}

```

File: src/base.jl [hunk 1]
```diff
@@ -270,26 +281,61 @@ Serializes a system to a JSON string.
...
 - `check::Bool = false`: whether to run system validation checks
 
 Refer to [`check_component`](@ref) for exceptions thrown if `check = true`.
 """
-function IS.to_json(sys::System, filename::AbstractString; force = false, runchecks = false)
+function IS.to_json(
+    sys::System,
+    filename::AbstractString;
+    user_data = nothing,
+    pretty = false,
```
Comment (daniel-thom) [inline src/base.jl:295]: This might only apply to me, but I like this addition.

**Review by claytonpbarrows — APPROVED** [review]
I really like this. I can't think of anything else that needs to be added.

**Review by jd-lara — APPROVED** [review]

File: src/base.jl [hunk 2]
```diff
@@ -270,26 +281,61 @@ Serializes a system to a JSON string.
...
 
+function _serialize_system_metadata_to_file(sys::System, filename, user_data)
+    name = get_name(sys)
+    description = get_description(sys)
+    resolution = Dates.Minute(get_time_series_resolution(sys)).value
+    metadata = OrderedDict(
+        "name" => isnothing(name) ? "" : name,
+        "description" => isnothing(description) ? "" : description,
+        "frequency" => sys.frequency,
+        "time_series_resolution_minutes" => resolution,
```
Comment (daniel-thom) [inline src/base.jl:324]: This reports the resolution as an integer and the units are in the field name. An alternative is like this:
```
"time_series_resolution": "60 minutes"
```
The string is probably easier to read. The integer is much easier to machine parse.

**Pull request merged**